hollow/src/main/java/com/netflix/hollow/tools/history/keyindex/HollowHistoryKeyIndex.java

```diff
@@ -148,21 +151,22 @@ public void run() {
 
     private HollowReadStateEngine roundTripStateEngine(boolean isInitialUpdate, boolean isSnapshot) {
         try {
+            Path tmpFile = Files.createTempFile("tmp", "history");
```

Comment:
Do something like `Files.createTempFile("roundtrip", "snapshot")`. This would make a file like `${java.io.tmpdir}/roundtripXXXXXX.snapshot`